crates/telio-traversal/src/endpoint_providers/upnp/mod.rs

```diff
@@ -132,47 +135,59 @@ impl UpnpEpCommands for IgdGateway {
             Some(gw) => gw,
             None => return Err(Error::NoIGDGateway),
         };
-        new_ext_port.0 = gw.add_any_port(
-            PortMappingProtocol::UDP,
-            SocketAddrV4::new(ip_addr, proxy_port),
-            3600,
-            "libminiupnp",
-        )?;
-        new_ext_port.1 = gw.add_any_port(
-            PortMappingProtocol::UDP,
-            SocketAddrV4::new(ip_addr, wg_port),
-            3600,
-            "libminiupnp",
-        )?;
+        new_ext_port.0 = gw
+            .add_any_port(
+                PortMappingProtocol::UDP,
+                SocketAddrV4::new(ip_addr, proxy_port),
+                3600,
+                "libminiupnp",
+            )
+            .await?;
+        new_ext_port.1 = gw
+            .add_any_port(
+                PortMappingProtocol::UDP,
+                SocketAddrV4::new(ip_addr, wg_port),
+                3600,
+                "libminiupnp",
+            )
+            .await?;
         Ok(new_ext_port)
     }
 
-    fn delete_endpoint_routes(&self, proxy_port: u16, wg_port: u16) -> Result<()> {
+    async fn delete_endpoint_routes(&self, proxy_port: u16, wg_port: u16) -> Result<()> {
         let gw = match &self.gw {
             Some(gw) => gw,
             None => return Err(Error::NoIGDGateway),
         };
-        gw.remove_port(PortMappingProtocol::UDP, wg_port)?;
-        gw.remove_port(PortMappingProtocol::UDP, proxy_port)?;
+        gw.remove_port(PortMappingProtocol::UDP, wg_port).await?;
+        gw.remove_port(PortMappingProtocol::UDP, proxy_port).await?;
         Ok(())
     }
 
-    fn get_igd_gateway(&mut self) -> Result<()> {
-        self.gw = Some(search_gateway(Default::default())?);
+    async fn search_igd_gateway(&mut self) -> Result<()> {
+        self.gw = Some(Box::pin(search_gateway(Default::default())).await?);
```

Comment:
Why is this inner future pinned ?

Reply:
clippy complains about having a future too large